docs/asciidoc/faq.adoc

```diff
@@ -205,5 +226,113 @@ processor.publishOn(scheduler1)
          .doOnNext(i -> processNext(i))
          .subscribe();
 ----
+====
+
+== What Is a Good Pattern for Contextual Logging? (MDC)
+As described in section <<context>>, `ThreadLocal` does not make much sense in Reactive Streams, where several streams could run steps on a common thread.
+Instead of using `ThreadLocal`, Reactor offers the `Context`, a map-like object that is tied to a `Subscription` and not a `Thread`.
+
```

Comment:
Perhaps it would be more helpful to describe this in the context of a functional, declarative API rather than Reactive Streams? Those are meant to be thread agnostic, and each operator/stage could execute in a different thread. Then it's also easier to talk about passing a context through the composition chain as a more idiomatic solution in a functional API, and this is why Reactor has a built-in `Context` that propagates throughout the execution chain as long as Flux or Mono is used as the return value, etc.

So that takes you pretty far, but within a reactive component you might still end up calling something that doesn't have reactive types in its signature and for MDC logging context might become a necessary solution.

In other words perhaps worth phrasing it in such a way as to emphasize that MDC shouldn't be an automatic assumption, something that is all too easy to assume.

Reply:
reworked the introduction to address this :+1: